### PR TITLE
Implement RFC0006

### PR DESCRIPTION
**Pull request by kylef**
This pull request implements RFC0006 ([File Source Map](https://github.com/refractproject/rfcs/blob/master/text/0000-file-source-map.md)).

@zdne Would you be able to review this and ensure it matches your intent with the RFC.

### CODE REVIEW

**Comment by zdne** [comment]
One comment, otherwise looks good to me.

File: namespaces/parse-result.md [hunk 1]
```diff
@@ -104,12 +109,17 @@ Annotation for a source file. Usually generated by a parser or adapter.
...
-The Source Map element MAY be used in any other element attributes in its
-normal ("unrefracted") form under the `sourceMap` property.
+Every refract element MAY include a `sourceMap` attribute. Its content SHOULD
+be an array of `Source Map` elements. The Source Map elements represent the
+location(s) in source file(s) from which the element was composed.
 
 If used, it represents the location of characters in the source file.
 This location SHOULD include the characters used to build the parent element.
 
+The Source Map element SHOULD NOT be used it its normal ("unrefracted") form
```
Comment (pksunkara) [inline namespaces/parse-result.md]: MUST NOT

**Comment by kylef** [comment]
Feedback has been addressed.

File: namespaces/parse-result.md [hunk 2]
```diff
@@ -94,8 +94,13 @@ Annotation for a source file. Usually generated by a parser or adapter.
...
     "code": 6,
-    "sourceMap": [ [ 0, 10 ] ]
-  },
+    "sourceMap": [
+      {
+        "element": "sourceMap",
+        "content": [[4, 12], [20, 12]]
+      }
+    ]
+  }
```
Comment (pksunkara) [inline namespaces/parse-result.md]: Missing a `,` here

Comment (kylef) [inline namespaces/parse-result.md]: Added.

**Comment by pksunkara** [comment]
I can merge this, but that missing comma disrupts syntax highlighting in the mark-down file.

**Pull request merged**